torch/csrc/profiler/unwind/unwind.cpp

```diff
@@ -340,6 +341,26 @@ struct Symbolizer {
   }
   static Symbolizer& get() {
     static Symbolizer singleton;
+    if (addr2line_binary == nullptr) {
+        auto envar = std::getenv("TORCH_ADDR2LINE_BINARY");
+        if (envar != nullptr) {
+          // we require users to specify the full path to binary when using this
+          // flag for simplicity currently
+          struct stat buffer;
+          if(stat(envar, &buffer) == 0) {
```

Comment:
There are a lot of ways this binary might not work (mightn't be a file, might not be executable, might not exist). But this only checks for one. It might be preferable to just let the call to addr2line fail when invoked rather than try to enumerate all possible failures here.

Reply:
Yeah, I was also wondering what's the best expectation of this "precheck" step when writing this part. Also I am wondering if we should do this limited scope precheck in the very beginning of the torch initialization phase, because I want to best mitigate the case that after a long run in the end when we have something interesting to dump only to find out we had a typo in the binary path which ruined the chance to collect info. I was originally planning to fail the execution completely on the spot to make it "fail fast" instead of the current falling back to the default addr2line with a few warning lines. Curious to see your take here.